base/int.jl

```diff
@@ -305,6 +305,7 @@ promote_rule{T<:BitSigned64}(::Type{UInt64}, ::Type{T}) = UInt64
 promote_rule{T<:Union{UInt32, UInt64}}(::Type{T}, ::Type{Int128}) = Int128
 promote_rule{T<:BitSigned}(::Type{UInt128}, ::Type{T}) = UInt128
 
+promote_op{T<:Integer}(op, ::Type{T}) = typeof(op(one(T)))
```

Comment:
This series of promotion rules can also be removed from this file and applied to all `Number` types I think.